cis/processor.py

```diff
@@ -0,0 +1,127 @@
+"""Iterative Stream Processor plugin for handling steps during data storage."""
+import json
+import logging
+import os
+
+from cis import user
+
+from cis.libs import encryption
+from cis.libs import streams
+from cis.libs import validation
+
+logger = logging.getLogger(__name__)
+
+class OperationDelegate(object):
+    def __init__(self, boto_session=None, publisher=None, signature=None, encrypted_profile_data=None):
+        self.boto_session = boto_session
+        self.dry_run = True
+        self.decryptor = encryption.Operation(boto_session=boto_session)
+        self.encrypted_profile_data = encrypted_profile_data
+        self.kinesis_client = None
+        self.publisher = publisher
+        self.signature = signature
+        self.stage = self._get_stage()
+        self.user = None
+
+    def run(self):
+        # Determine what stage of processing we are in and call the corresponding functions.
+
+        self.decrytped_profile = json.loads(self._decrypt_profile_packet())
+
+        if self.stage == 'validator':
+            logger.info('Validator logic activated.')
+            res = self._validator_stage()
+
+            if res is True:
+                logger.info('Payload valid sending to kinesis.')
+                res = streams.Operation(
+                    boto_session=self.boto_session,
+                    publisher=self.publisher,
+                    signature=self.signature,
+                    encrypted_profile_data=self.encrypted_profile_data
+                ).to_kinesis()
+
+                if res['ResponseMetadata']['HTTPStatusCode'] == 200:
+                    res = True
+        elif self.stage == 'streamtoidv':
+            res = self._vault_stage()
+        elif self.stage == 'idvtoauth0':
+            # TBD fold in the authzero logic from idvtoauth0 in cis_functions
+            pass
+        else:
+            # Unhandled pass for anything not handled in block.  Basically yield to block.
+            res = None
+
+        logger.info('The result of the change was {r}'.format(r=res))
+        return res
+
+    def _decrypt_profile_packet(self):
+        return self.decryptor.decrypt(
+            ciphertext=self.encrypted_profile_data.get('ciphertext'),
+            ciphertext_key=self.encrypted_profile_data.get('ciphertext_key'),
+            iv=self.encrypted_profile_data.get('iv'),
+            tag=self.encrypted_profile_data.get('tag')
+
+        )
+
+    def _get_stage(self):
+        # Let the object know what phase of operation we are running in.
+        stage = os.environ.get('APEX_FUNCTION_NAME', None)
```

Comment:
I'm going to leave this here for now.  But I agree with you at a functional level.  The two should be independent of one another. Let's reach an agreement on how CIS should identify whether it's running in a stream processor in lambda or not.  

FYI I don't see a future where we have stream processors that don't run in lambda.